src/test/java/com/googlecode/objectify/test/EmbeddedNullContainerTests.java

```diff
@@ -1,2 +1,69 @@
-package com.googlecode.objectify.test;public class EmbeddedNullContainerTests {
+package com.googlecode.objectify.test;
+
+import com.googlecode.objectify.annotation.Entity;
+import com.googlecode.objectify.annotation.Id;
+import com.googlecode.objectify.test.util.TestBase;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.junit.jupiter.api.Test;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.googlecode.objectify.ObjectifyService.factory;
+
+public class EmbeddedNullContainerTests extends TestBase {
+
+  @Test
+  void testLoadingEmbeddedMapNullValue() {
+    factory().register(Sample.class);
+
+    List<Value> valueList = new ArrayList<>();
+    valueList.add(new Value(null, null));
+
+    Map<String, List<Value>> values = new HashMap<String, List<Value>>();
+    values.put("k1", valueList);
+
+    final Sample sample = new Sample("testKey", values);
+    final Sample retrieved  = saveClearLoad(sample);
+
+    assertThat(sample.values).isEqualTo(retrieved.values);
+  }
+
+
+  @Test
+  void testLoadingNullListValue() {
+    factory().register(Sample.class);
+
+    Map<String, List<Value>> values = new HashMap<String, List<Value>>();
+    values.put("k1", null);
```

Comment:
@stickfigure the issue seems to be that in the translated Entity, `values` is being interpreted as a map (v/s a list) with no elements.

IIUC, the correct behavior would be to interpret `values` as a list with a list element Entity `{"k1":null}`

Reply:
@eddavisson can you take a look while we try to get in touch with @stickfigure ?

Reply:
> `values` is being interpreted as a map (v/s a map) with no elements.
Didn't quite follow this.

The approach seems reasonable, but I don't know this code base well; not sure if there are other implications to making this change.

Does `MapifyTranslatorFactory` need the same adjustment?

Reply:
Sorry, updated the comment to make more sense.

Reply:
As for `MapifyTranslatorFactory`, I'm taking my cue from the comment 
```
// If the collection does not exist, skip it entirely. This mirrors the OLD underlying behavior
// of collections in the datastore; if they are empty, they don't exist.
```

Reply:
I did base the fix on `v6` as the Customer hit this issue on v6. Hopinh we'd want to patch a fix on top of it and not on `main` which may have some unfinished/unreleased functionality, I based the fix on v6.

As for `testLoadingNullListValue`, the assert fails for me at this version - I've added sample output in the description of the commit.

Reply:
1. Ok, i'm able to reproduce the original issue at `master` too for `EmbeddedMapTranslatorFactory`. See the attached test file `App.java` which I used to reproduce using `testLoadingNullEmbeddedMapValueUsingObjectifyV6` methid in `App.java`. I'm also able to confirm that my fix works for this issue.

2. The same issue for `CollectionTranslatorFactory` seems to have been `fixed` earlier by adding the check for `node.get() == null` and throwing `SkipException` but that seems to be the incorrect fix - the right thing to do there too is to `return null`. I used `testLoadingEmbeddedMapNullValue` to verify this.


```
package com.googlecode.objectify.test.integration;

import static org.junit.Assert.assertEquals;

import com.google.cloud.datastore.DatastoreOptions;
import com.googlecode.objectify.Key;
import com.googlecode.objectify.ObjectifyFactory;
import com.googlecode.objectify.ObjectifyService;
import com.googlecode.objectify.annotation.Entity;
import com.googlecode.objectify.annotation.Id;
import com.googlecode.objectify.util.Closeable;
import java.io.PrintWriter;
import java.io.StringWriter;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

public class App {

  @Entity(name = "Sample")
  @Data
  @NoArgsConstructor
  @AllArgsConstructor
  public static class Sample {
    @Id String name;
    Map<String, List<Value>> values;
  }

  @Data
  @NoArgsConstructor
  @AllArgsConstructor
  public static class Value {
    String primitiveField;
    Map<String, String> structuredField;
  }

  public static void testLoadingNullEmbeddedMapValueUsingObjectifyV6() {
    List<Value> valueList = new ArrayList<>();
    valueList.add(new Value(null, null));

    Map<String, List<Value>> values = new HashMap<String, List<Value>>();
    values.put("k1", valueList);

    String appDocName = "sample1"; // entered from app
    String consoleDocName = "sample2"; // entered from console, both null
    //Sample consoleDocSample = new Sample(consoleDocName, values);
    Sample appDocSample = new Sample(appDocName, values);

    DatastoreOptions datastoreOptions = DatastoreOptions.newBuilder()
        .setProjectId("jimit-test")
        .setNamespace("objectify-test")
        .build();
    ObjectifyFactory factory = new ObjectifyFactory(datastoreOptions.getService());
    factory.register(Sample.class);
    ObjectifyService.init(factory);

    try (Closeable ignored = ObjectifyService.begin()) {
      // If an entity is not saved, uncomment the following line and call save() method.
      // And edit the entity directly to `structuredFiled` has null (nullValue) from GCP console
      factory.ofy().save().entity(appDocSample).now();

      Sample appDocEntity = factory.ofy().load().key(Key.create(Sample.class, appDocName)).now();
      Sample consoleDocEntity = factory.ofy().load().key(Key.create(Sample.class, consoleDocName))
          .now();

      System.out.println("appDoc=" + appDocEntity);
      System.out.println("consoleDoc=" + consoleDocEntity);
      assertEquals(appDocEntity, consoleDocEntity);
    } catch (Exception e) {
      StringWriter sw = new StringWriter();
      e.printStackTrace(new PrintWriter(sw));
      System.out.print(sw);
    }
  }


    public static void testLoadingEmbeddedMapNullValue() {
      Map<String, List<Value>> values = new HashMap<>();
      values.put("k1", null);

      String appDocId = "sample3"; // entered from test app
      String consoleDocId = "sample4"; // entered from console, both {k1 = null}
      Sample appDoc = new Sample(appDocId, values);

      DatastoreOptions datastoreOptions = DatastoreOptions.newBuilder()
          .setProjectId("jimit-test")
          .setNamespace("objectify-test")
          .build();
      ObjectifyFactory factory = new ObjectifyFactory(datastoreOptions.getService());
      factory.register(Sample.class);
      ObjectifyService.init(factory);

        try (Closeable ignored = ObjectifyService.begin()) {
          // If an entity is not saved, uncomment the following line and call save() method.
          // And edit the entity directly to `structuredFiled` has null (nullValue) from GCP console
          factory.ofy().save().entity(appDoc).now();

          Sample appDocEntity = factory.ofy().load().key(Key.create(Sample.class, appDocId)).now();
          Sample consoleDocEntity = factory.ofy().load().key(Key.create(Sample.class, consoleDocId)).now();

          System.out.println("appDocEntity=" + appDocEntity.toString());
          System.out.println("consoleDocEntity=" + consoleDocEntity.toString());

          assertEquals(appDocEntity, consoleDocEntity);
        } catch (Exception e) {
             StringWriter sw = new StringWriter();
             e.printStackTrace(new PrintWriter(sw));
             System.out.print(sw);
    }
  }

  public static void main(String[] args) {

    testLoadingNullEmbeddedMapValueUsingObjectifyV6();
    testLoadingEmbeddedMapNullValue();
  }
}
```